In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('data/titanic_data.csv')

In [3]:
df.dropna()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
1           1       1  female  38.0      1      0  71.2833        C  First   
3           1       1  female  35.0      1      0  53.1000        S  First   
6           0       1    male  54.0      0      0  51.8625        S  First   
10          1       3  female   4.0      1      1  16.7000        S  Third   
11          1       1  female  58.0      0      0  26.5500        S  First   
..        ...     ...     ...   ...    ...    ...      ...      ...    ...   
869         1       1  female  47.0      1      1  52.5542        S  First   
870         0       1    male  33.0      0      0   5.0000        S  First   
877         1       1  female  56.0      0      1  83.1583        C  First   
885         1       1  female  19.0      0      0  30.0000        S  First   
887         1       1    male  26.0      0      0  30.0000        C  First   

       who  adult_male deck  embark_town alive  alone  
1    woman       False    C    Cherbourg   yes  False  
3    woman       False    C  Southampton   yes  False  
6      man        True    E  Southampton    no   True  
10   child       False    G  Southampton   yes  False  
11   woman       False    C  Southampton   yes   True  
..     ...         ...  ...          ...   ...    ...  
869  woman       False    D  Southampton   yes  False  
870    man        True    B  Southampton    no   True  
877  woman       False    C    Cherbourg   yes  False  
885  woman       False    B  Southampton   yes   True  
887    man        True    C    Cherbourg   yes   True  

[182 rows x 15 columns]

In [4]:
df.shape

(889, 15)

In [5]:
df.size

13335

In [6]:
df.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'class', 'who', 'adult_male', 'deck', 'embark_town',
       'alive', 'alone'],
      dtype='object')

<hr>

## What is the total number of passengers from each class who survived?

In [7]:
df1 = df.groupby('class').sum()

C:\Users\Illum\AppData\Local\Temp\ipykernel_26240\1515836357.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df1 = df.groupby('class').sum()


In [8]:
df1[['survived']]

survived
class           
First        136
Second        87
Third        119

#### 136 people survived in First Class, 87 in Second Class, and 119 in Third Class

<hr>

## What was the average fare for passengers from each town?

In [9]:
df2 = df.groupby('embark_town').mean()

C:\Users\Illum\AppData\Local\Temp\ipykernel_26240\3937750091.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df2 = df.groupby('embark_town').mean()


In [10]:
df2[['fare']]

fare
embark_town           
Cherbourg    59.954144
Queenstown   13.348741
Southampton  27.109647

#### 59.95 was the average fare in Cherbourg, 13.35 in Queenstown, and 27.11 in Southampton.</p>

<hr>

## What was the average fare for male vs female passengers from each town?

In [14]:
df3 = df.groupby(['embark_town', 'sex']).mean()

C:\Users\Illum\AppData\Local\Temp\ipykernel_26240\521349617.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df3 = df.groupby(['embark_town', 'sex']).mean()


In [15]:
df3[['fare']]

fare
embark_town sex              
Cherbourg   female  75.169805
            male    48.262109
Queenstown  female  12.634958
            male    13.991145
Southampton female  38.740929
            male    21.743397

#### In Cherbourg, females averaged 75.17 while males averaged 48.26.
#### In Queenstown, females averaged 12.63 while males averaged 13.99.
#### In Southampton, females averaged 38.74 while males averaged 21.74.

<hr>

## What was the average fare and age for male vs female passengers from each town?

In [16]:
df3[['fare', 'age']]

fare        age
embark_town sex                         
Cherbourg   female  75.169805  28.344262
            male    48.262109  32.998841
Queenstown  female  12.634958  24.291667
            male    13.991145  30.937500
Southampton female  38.740929  27.771505
            male    21.743397  30.292234

#### Females in Cherbourg averaged 75.17 at 28 years old, while the males averaged 48.26 at 33 years old.
#### Females in Queenstown averaged 12.63 at 24 years old while the males averaged 13.99 at 31 years old.
#### Females in Southampton averaged 38.74 at 28 years old while the males averaged 21.74 at 30 years old.